build.psm1

```diff
@@ -723,7 +723,24 @@ function New-PSOptions {
         $Top = [IO.Path]::Combine($PSScriptRoot, "src", "System.Management.Automation")
     }
 
-    return @{ Top = $Top;
+    $RootInfo = @{RepoPath = $PSScriptRoot}
+
+    # the valid root is the root of the filesystem and the folder PowerShell
+    $RootInfo += @{ValidPath = Join-Path -Path ([system.io.path]::GetPathRoot($RootInfo.RepoPath)) -ChildPath 'PowerShell' }
+    
+    if($RepoInfo.RepoPath -ne $RootInfo.ValidPath)
+    {
+        $RootInfo += @{Warning = "Please ensure you repo is at the root of the file system and named 'PowerShell' (example: '$($RootInfo.ValidPath)'), when building and packaging for release!" }
+        $RootInfo += @{IsValid = $false}
+        Write-Warning -Message $RootInfo.Warning
```

Comment:
Don't write the warning in `New-PSOptions`. We only care about the root path when it comes to a release build.